src/frontend/src/lib/services/batch.services.ts

```diff
@@ -0,0 +1,30 @@
+export const batchPromisesPerSecond = async function* <T, R>({
+	items,
+	maxCallsPerSecond,
+	promise,
+	validate,
+	afterPromise
+}: {
+	items: T[];
+	maxCallsPerSecond: number;
```

Comment:
I did not understood `maxCallsPerSecond` from the naming. Now that I read the code I get it but, it was not natural. Any other naming idea?

Reply:
I went with `batchSize` ? WDYT